CIP-0137/README.md

```diff
@@ -413,97 +425,169 @@ stateDiagram-v2
   StBusy:::Green --> StIdle:::Blue : MsgAcceptMessage
   StBusy:::Green --> StIdle:::Blue : MsgRejectMessage
   StIdle:::Blue --> StDone:::Black : MsgDone
-   
+
 ```
 
 ##### Protocol messages
 
-* **MsgSubmitMessage(message)**: The client submits a message.
-* **MsgAcceptMessage**: The server accepts the message.
-* **MsgRejectMessage(reason)**: The server rejects the message and replies with a *reason*.
-* **MsgDone**: The client terminates the mini-protocol.
+- **MsgSubmitMessage(message)**: The client submits a message.
+- **MsgAcceptMessage**: The server accepts the message.
+- **MsgRejectMessage(reason)**: The server rejects the message and replies with a _reason_.
+- **MsgDone**: The client terminates the mini-protocol.
 
 ##### Transition table
 
-| From state | Message | Parameters | To State|
-|--------|------|----|----| 
-| StIdle | MsgSubmitMessage | message | StBusy |
-| StBusy | MsgAcceptMessage |  | StIdle |
-| StBusy | MsgRejectMessage | reason | StIdle |
-| StIdle | MsgDone |  | StDone |
+| From state | Message          | Parameters | To State |
+| ---------- | ---------------- | ---------- | -------- |
+| StIdle     | MsgSubmitMessage | message    | StBusy   |
+| StBusy     | MsgAcceptMessage |            | StIdle   |
+| StBusy     | MsgRejectMessage | reason     | StIdle   |
+| StIdle     | MsgDone          |            | StDone   |
 
+##### CDDL Encoding Specification
+
+```cddl
+localMessageSubmissionMessage
+  = msgSubmitMessage
+  / msgAcceptMessage
+  / msgRejectMessage
+  / msgDone
+
+msgSubmitMessage = [0, message]
+msgAcceptMessage = [1]
+msgRejectMessage = [2, reason]
+msgDone          = [3]
+
+reason = invalid
+       / alreadyReceived
+       / ttlTooLarge
+       / other
+
+invalid         = [0, tstr]
+alreadyReceived = [1]
+ttlTooLarge     = [2]
+other           = [3, tstr]
+
+messageId    = bstr
+messageBody  = bstr
+blockNumber  = word32
+ttl          = word16
+kesSignature = bstr
+operationalCertificate = bstr
+
+message = [
+  messageId,
+  messageBody,
+  blockNumber,
+  ttl
+  kesSignature,
+  operationalCertificate
+]
+```
 
 ### Local Message Notification mini-protocol
 
 #### Description
 
-The local message notification mini-protocol is used by local clients to be notified about new message received by the network node. 
+The local message notification mini-protocol is used by local clients to be notified about new message received by the network node.
 
 The protocol follows a simple request-response pattern:
 
 1. The client sends a request with a single message.
-2. The server either accepts the message (and returns a confirmation) or rejects it (and returns the reason)
+2. The server either has messages that it can provide, returning the list of all available messages; or doesn't have any, in which case it returns a suitable response saying exactly that.
```

Comment:
This will lead to burst of data; but this might be unavoidable if signatures are created roughly at the same time around the system.

How much data do you expect to be sent at once?

Reply:
We expect at most `3000 messages * 2.8KB < 10MB` so the burst is not that important

Reply:
There's more rationale about this decisions on this document: https://github.com/IntersectMBO/ouroboros-network/wiki/Decentralized-Message-Queue-(DMQ)-Implementation-Overview

Reply:
ok, thanks